app/models/report_template.rb

```diff
@@ -40,8 +40,9 @@ def taxonomy_foreign_conditions
     { :report_template_id => id }
   end
 
-  def suggested_report_name
-    "#{name}-#{Date.today}.txt"
+  def suggested_report_name(type)
+    type ||= "csv"
+    "#{name}-#{Date.today}.#{type}"
```

Comment:
shouldnt this change the content html header? e.g. something like 'application/csv' or similar?